crates/deltalake-core/src/table/mod.rs

```diff
@@ -448,6 +449,23 @@ impl DeltaTable {
         self.update().await
     }
 
+    /// Get the commit obj from the version
+    pub async fn get_obj_from_version(
+        &self,
+        current_version: i64,
+    ) -> Result<Bytes, DeltaTableError> {
+        let commit_log_bytes = match self.log_store.read_commit_entry(current_version).await {
+            Ok(bytes) => Ok(bytes),
+            Err(DeltaTableError::ObjectStore {
+                source: ObjectStoreError::NotFound { .. },
+            }) => {
+                return Err(DeltaTableError::DeltaLogNotFound(current_version));
+            }
+            Err(err) => Err(err),
```

Comment:
Why return `Ok(Err(err))` instead of `Err(err)`?

Reply:
Sorry, where is `Ok(Err(err))`?
Here we use the same method as `peek_next_commit` below to get the `commit_log_bytest`, then we return the full `commit_log_bytes`.

Reply:
In this line, we assign `commit_log_bytes` to `Err(err)`. Then on line 470, we return `Ok(commit_log_bytes)`. I believe as a result we will return `Ok(Err(err))` if we reach line 468.

Am I interpreting this correctly? If so, this seems wrong. I think you'd want to `return Err(err)` here and on line 462 you'd want to assign `commit_log_bytes` to `bytes`, not `Ok(bytes)` (which would thus return `Ok(Ok(bytes))` which is weird as well).

Reply:
yes seems weird we need to fix this, also need to add a test for error case.